qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+            queue_time=queue_time,
+            status=status,
+        )
+
+    def get_device(self, device_id: str) -> dict[str, Any]:
+        """Get a specific Azure device."""
+        data = self.session.get_device(device_id)
+        profile = self._build_profile(data)
+        return AzureQuantumDevice(profile, self.session)
+
+    def get_devices(self, **kwargs) -> list[AzureQuantumDevice]:
+        """Get a list of Azure devices."""
+        devices = self.session.get_devices()
+
+        quantum_devices = [
+            AzureQuantumDevice(self._build_profile(device), self.session) for device in devices
+        ]
+        return quantum_devices
+
+
+class AzureHelperFunctions:
+    """Helper functions for Azure Quantum API."""
+
+    def quantum_access_token(
+        self, client_id: str, client_secret: str, tenant_id: str, session: Any
+    ) -> str:
+        """Generates access token for Azure Quantum API."""
+        data = {
+            "grant_type": "client_credentials",
+            "client_id": client_id,
+            "client_secret": client_secret,
+            "scope": "https://quantum.microsoft.com/.default",
+        }
+
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        url = f"https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token"
+        r = session.post(url, headers=headers, data=data, verify=True)
+        quantum_access_token = r.json()["access_token"]
+
+        return quantum_access_token
+
+    def storage_access_token(
+        self, client_id: str, client_secret: str, tenant_id: str, session: Any
+    ) -> str:
+        """Generates access token for Azure Storage API."""
+        data = {
+            "grant_type": "client_credentials",
+            "client_id": client_id,
+            "client_secret": client_secret,
+            "scope": "https://management.azure.com/.default",
+        }
+
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        url = f"https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token"
+        r = session.post(url, headers=headers, data=data, verify=True)
+        storage_access_token = r.json()["access_token"]
+
+        return storage_access_token
+
+    @staticmethod
+    def device_manager(devices: dict) -> dict[str, dict[str, Any]]:
+        """Manages the data for all Azure Quantum devices."""
+        devices_dict = {}
+
+        for provider in devices["value"]:
+            for machine in provider["targets"]:
+                if machine["id"][:4] != "ionq":
+                    devices_dict[machine["id"]] = {
+                        "status": machine["currentAvailability"],
+                        "isAvailable": machine["currentAvailability"] == "Available",
+                        "nextAvailable": None,
+                        "availablilityCD": "",
+                        "averageQueueTime": machine["averageQueueTime"],
+                    }
+
+        return devices_dict
```

Comment:
status and availability data should be dynamically retrieved from methods within the device class, not statically loaded. Also, you do not need to conform the format of the dictionary to our API schema. Our schemas can and will change. Uses the data structures that make the most sense in the context.